quadcontrol/scenario_parser.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Phoenix-Pilot
+ *
+ * Flight scenario parser
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Piotr Nieciecki
+ *
+ * This file is part of Phoenix-Pilot software
+ *
+ * %LICENSE%
+ */
+
+#include "scenario_parser.h"
+#include <parser.h>
+
+#include <stdlib.h>
+
+
+#define INIT_SIZE 5
+
+
+static int invCnt;
+
+static struct {
+	flight_mode_t *result;
+	int invCnt; /* number of converters invocations */
+} scenarioParser_common;
+
+
+int quad_scenarioRead(flight_mode_t **scenario, const char *path)
```

Comment:
I suggest to use convention, that the last arguments are the output args.

```suggestion
int quad_scenarioRead(const char *path, flight_mode_t **scenario, size_t *sz)
```